src/python/WMCore/FwkJobReport/Report.py

```diff
@@ -599,16 +601,11 @@ def addError(self, stepName, exitCode, errorType, errorDetails, siteName=None):
         errDetails.type = str(errorType)
 
         try:
-            if hasattr(errorDetails, "decode"):
-                # Fix for the unicode encoding issue, #8043
-                # interprets this string using utf-8 codec and ignoring any errors
-                errDetails.details = errorDetails.decode('utf-8', 'ignore')
-            else:
-                # Then cast it to string and decode it
-                errorDetails = str(errorDetails)
-                errDetails.details = errorDetails.decode('utf-8', 'ignore')
+            errDetails.details = newstr(errorDetails)
```

Comment:
Dario, can you educate me here why we no longer need to decode the "text" provided in the errorDetails? Would it work with those weird bytes/words mentioned in the previous GH issues (like `0xff` or `\x95`)?

This code has given us so much headache in the past, that I'm very concerned of having regressions.

Reply:
I do feel your headache! The original discussion was in #8403

What you are doing here is

```plaintext
if errorDetails is bytes or unicode 
  # however only bytes should have decode attribute
  # if errorDetails is unicode and it contains only ascii, then it is decoded back to itself
  errDetails.details = unicode(errorDetails)
else
  # here we can have dict, list, ...
  errorDetails = bytes(errorDetails)
  errDetails.details = unicode(errorDetails)
```

My approach would be

```plaintext
errDetails.details = unicode(errorDetails)
```

Regarding malformed bytes, you are correct. Those would break `newstr(errorDetails)`, but that is why we have the try/catch clause! However, now that I stare at it, I am puzzled. It catches "UnicodeEncodeError", but the message states "Failed to **decode** the job error details for job ID: %s.". 

Maybe you are correct, I should give this another try!

Reply:
The plot thickens!

I opted for

```python
from builtins import str as newstr, bytes
...
        try:
            if isinstance(errorDetails, newstr):
                errDetails.details = errorDetails
            elif isinstance(errorDetails, bytes):
                errDetails.details = errorDetails.decode('utf8', 'ignore')
            else:
                errDetails.details = newstr(errorDetails)
``` 

but even if `errorDetails` were py2-`str` (here also `bytes`), then it would fail with 
> UnicodeEncodeError: 'ascii' codec can't encode characters in position 2-5: ordinal not in range(128)
which made no sense because I was trying to decode, not encode bytes.

I after a full hour, I removed the `try` and realized that it was failing here https://github.com/dmwm/WMCore/blob/595a9934aef084b3aaf5ca93c55a145c405edb43/src/python/WMCore/Configuration.py#L141. The bytes was successfully converted to unicode, but then was converted back to bytes with ascii encoding by `configuration.py`

I then decided to change `Configuration` a bit to ease or near-future work with using `Utils.Utilities.encodeUnicodeToBytes()`. This is a temporary fix, we will likely be better of changing this when switching to py3.

Moreover this raises a question: do we really need to decode to unicode in `addError`, if then it would be converted to bytes anyway by `Configuration`?